Use eleventy to statically generate protocol docs

TimvdLippe opened the pull request:
This updates the protocol viewer to use Eleventy to statically generate the docs. Since we don't have any actual interactive elements, the usage of web components made little sense for these parts. The search functionality remain web components, as these are by nature dynamic.

It has been a lot of tweaking and the commit history shows the overall process. However, for reviewing I suggest locally building and checking out the output on http://localhost:8000/devtools-protocol/

E.g. `npm i && npm run build && open http://localhost:8000/devtools-protocol/`

Some features:
1. Statically generate, achieving 100 on LightHouse performance
1. Removed a lot of unnecessary JavaScript and significantly reduced the overall size of the website
1. Use flex-grid for parameters section for improved responsiveness (woohoo!)
1. Rewrite search components to use lit instead of Polymer
1. Fix references to hashes. E.g. http://localhost:8000/devtools-protocol/tot/Profiler/#method-disableRuntimeCallStats will now correctly deeplink again
1. Typing any alphanumerical character will automatically start the search.
1. Use markdown for the index page, rather than HTML for easier editing.
1. Use rollup with terser for a small JavaScript build.
1. Ensure that tabbing throughout the page is consistent, including showing the pound signs when inspecting a method/event/type
1. Small accessibility tweaks in terms of color contrast

CC @paullewis

Closes #109
Fixes #99

TimvdLippe commented:
1 annoyance: If the user is selects a result from a search that is on the same page, then the page doesn't properly scroll into the view. I haven't been able to figure out how to force the page to scroll to the correct location.

TimvdLippe commented:
And just as I posted it and did some more debugging I figured it out: I was listening for keydown instead of keyup. As such, the value I was reading from the input element was the old value, not the new value. Now that should all be working again 🎉

paulirish commented:
wow. this is pretty awesome. and fast.

god, static sites are so beautiful.

3 things i noticed so far.  but yeah i'm pretty on board with merging this. feels good.

----------

> 1 annoyance: If the user is selects a result from a search that is on the same page, then the page doesn't properly scroll into the view. I haven't been able to figure out how to force the page to scroll to the correct location.

scrollIntoViewIfNeeded() with a scrollIntoView() fallback ?


nice keyboard fuzzysearch.. but small bug on the event capturing. just tried `cmd-up` for pageup and the input field intercepted on `cmd` and broke it.

in search results, can events get prefixed with their domain?

TimvdLippe commented:
> wow. this is pretty awesome. and fast.
> 
> god, static sites are so beautiful.
> 
> 3 things i noticed so far. but yeah i'm pretty on board with merging this. feels good.
> 
> > 1 annoyance: If the user is selects a result from a search that is on the same page, then the page doesn't properly scroll into the view. I haven't been able to figure out how to force the page to scroll to the correct location.
> 
> scrollIntoViewIfNeeded() with a scrollIntoView() fallback ?

I will defer this one to a follow-up CL, as I need to do more investigation. It's a minor issue luckily.
> nice keyboard fuzzysearch.. but small bug on the event capturing. just tried `cmd-up` for pageup and the input field intercepted on `cmd` and broke it.

Fixed.
> in search results, can events get prefixed with their domain?

Fixed.

tophf commented on pages/scripts/index.js:
```diff
@@ -0,0 +1,332 @@
+
+document.addEventListener('keydown', (event) => {
+  // One of `a-z` or `A-Z`
+  if (event.keyCode >= 65 && event.keyCode <= 90) {
```
You need to add a check for modifiers like event.ctrlKey otherwise it's impossible to copy a selected fragment of text in the page via keyboard.

TimvdLippe replied:
That should be fixed in https://github.com/ChromeDevTools/debugger-protocol-viewer/commit/78c0fc679cddbc5070084d3a8814608326af9e2a

It is working for me, is it not working for you? If so, please open a new issue.

tophf replied:
metaKey is not ctrlKey so that won't work in Windows...

TimvdLippe replied:
Ah of course, you are absolutely correct. Do you mind sending a PR with a fix or should I fix it?

tophf replied:
Better if you do it. Maybe also skip altKey but it needs testing with AltGr keyboards, I just don't know, never used one myself.
